backend/src/main/java/gov/cdc/usds/simplereport/service/OktaServiceImpl.java

```diff
@@ -123,25 +147,37 @@ public void deleteOrganization(String externalId) {
     }
 
     // returns the external ID of the organization the specified user belongs to
-    public String getOrganizationExternalIdForUser(String username) {
+    public Optional<OrganizationRoleClaims> getOrganizationRolesForUser(String username) {
+
         UserList users = _client.listUsers(username, null, null, null, null);
         if (!users.iterator().hasNext()) {
             throw new IllegalGraphqlArgumentException("Cannot get org external ID for nonexistent user");
         }
         User user = users.single();
 
+        Set<String> orgExternalIds = new HashSet<>();
+        Set<OrganizationRole> roles = new HashSet<>();
+
         for (Group g : user.listGroups()) {
             String groupName = g.getProfile().getName();
-            // We assume that a user is only a member of one user group
-            if (g.getType() == GroupType.OKTA_GROUP &&
-                    groupName.startsWith(_rolePrefix) &&
-                    groupName.endsWith(generateRoleSuffix(OrganizationRole.USER))) {
-                return getOrganizationExternalIdFromGroupName(groupName, 
-                                                              OrganizationRole.USER);
+            for (OrganizationRole role : OrganizationRole.values()) {
+                if (g.getType() == GroupType.OKTA_GROUP &&
+                            groupName.startsWith(_rolePrefix) &&
+                            groupName.endsWith(generateRoleSuffix(role))) {
+                    orgExternalIds.add(getOrganizationExternalIdFromGroupName(groupName, role));
+                    roles.add(role);
+                    break;
+                }
             }
         }
 
-        throw new IllegalGraphqlArgumentException("User is not in any Okta user groups");
+        if (orgExternalIds.size() != 1) {
+            LOG.warn("User is in {} Okta organizations, not 1", orgExternalIds.size());
```

Comment:
eek, there's no notion of archiving groups in Okta. Either they live on (which feels not good, because then do the old user-group relationships remain? and then does the OktaService always return all groups--existing and archived--that the user belongs to?) or we get rid of them. I'd prefer the latter, which means this code stays as-is. let me know if you feel strongly.

Reply:
I feel strongly that we shouldn't have two conflicting definitions of "valid" in different places in the app—I think there are reasonable justifications for the approach that I took originally (well, obviously I do), though I also see your point about cleaning up quickly—worst case this won't be relevant for a significant while, so let's talk about it next week.